configs/components/ruby-2.7.8.rb

```diff
@@ -170,6 +171,19 @@
     ]
   end
 
+  if(platform.name =~ /windowsfips-2016/)
+    # We need the below patch since during ruby build step for windowsfips-2016-x64 agent-runtime builds,
+    # the rbconfig.rb file that gets generated contains '\r' trailing character in 'ruby_version' config.
+    # We patch rbconfig.rb to remove the '\r' character.
+    # This patch has to run after the build step since rbconfig.rb is generated during the build step. 
+    # This is sort of a hacky way to do this. We need to find why the '\r' character gets appended to
+    # 'ruby_version' field in the future so that this patch can be removed - PA-6902.
+    pkg.add_source("#{base}/rbconfig_win.patch")
+    pkg.build do
+      ["TMP=/var/tmp /usr/bin/patch.exe --binary --strip=1 --fuzz=0 --ignore-whitespace --no-backup-if-mismatch < ../rbconfig_win.patch"]
```

Comment:
Why construct the patch command instead of using the built in functionality in vanagon for patching a component?

Reply:
Hey @joshcooper,
I think you mean doing
`pkg.apply_patch "#{base}/rbconfig_win.patch"`
instead of the raw command.

I did that earlier but then the apply_patch method adds the patch to 'ruby-patch' step. This fails since we are patching rbconfig file which is generated after the ruby-build step.

Reply:
Ah right makes sense